### ISSUE TITLE
Ref safety analysis does not properly handle readonly set methods

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#73872]
Consider the following code:

```csharp
ref struct S 
{
    internal Span<int> Span 
    {
        get => default;
        readonly set { }
    }

    readonly void M(Span<int> span) { }

    static void Test()
    {
        // safe to escape from method
        S local = new S();

        // stack alloc
        Span<int> span = [1, 2];

        // Error!
        local.Span = span;

        // Okay
        local.M(span);
    }
}
```

This compiler flags the `local.Span = span` assignment as a ref safety error. This is not actually an error though because the `set` accessor is `readonly` hence there is no way for `span` to be stored. From a ref safety standpoint it is no different than `S.M` which the compiler does not flag as an error.

This is a fairly low priority bug. A `readonly set` is a bit of a non-sensical operation but the language does allow it.
